packages/block-editor/src/components/block-list/use-block-props/use-selected-block-event-handlers.js

```diff
@@ -134,31 +117,98 @@ export function useEventHandlers( { clientId, isSelected } ) {
+				clone.style.visibility = 'hidden';
+
+				// Remove the id and leave it on the clone so that drop target
+				// calculations are correct.
+				const id = node.id;
+				node.id = null;
 
-				if ( document !== ownerDocument ) {
-					const frame = defaultView.frameElement;
-					if ( frame ) {
-						const rect = frame.getBoundingClientRect();
-						offset = { x: rect.left, y: rect.top };
+				let _scale = 1;
+
+				let parentElement = node;
+
+				while ( ( parentElement = parentElement.parentElement ) ) {
+					const { scale } =
+						defaultView.getComputedStyle( parentElement );
+					if ( scale && scale !== 'none' ) {
+						_scale = parseFloat( scale );
+						break;
 					}
 				}
 
-				// chip handle offset
-				offset.x -= 58;
+				let bgColor = 'transparent';
 
-				function over( e ) {
-					domNode.style.transform = `translate( ${
-						e.clientX + offset.x
-					}px, ${ e.clientY + offset.y }px )`;
+				parentElement = node;
+
+				while ( ( parentElement = parentElement.parentElement ) ) {
+					const { backgroundColor } =
+						defaultView.getComputedStyle( parentElement );
+					if (
+						backgroundColor &&
+						backgroundColor !== 'transparent' &&
+						backgroundColor !== 'rgba(0, 0, 0, 0)'
+					) {
+						bgColor = backgroundColor;
+						break;
+					}
 				}
 
-				over( event );
+				const inverted = 1 / _scale;
+
+				node.after( clone );
+
+				node.style.position = 'fixed';
+				node.style.top = `${ rect.top }px`;
+				node.style.left = `${ rect.left }px`;
+				node.style.width = `${ rect.width * inverted }px`;
+
+				const originX = event.clientX - rect.left;
+				const originY = event.clientY - rect.top;
+
+				// Scale everything to 200px.
+				const dragScale = rect.height > 200 ? 200 / rect.height : 1;
+
+				node.style.zIndex = '1000';
+				node.style.transformOrigin = '0 0';
+				node.style.transformOrigin = `${ originX }px ${ originY }px`;
+				node.style.transition = 'transform 0.2s ease-out';
+				node.style.transform = `scale(${ dragScale })`;
+				node.style.margin = '0';
+				node.style.opacity = '0.9';
+				node.style.backgroundColor = bgColor;
+
+				let hasStarted = false;
+
+				function over( e ) {
+					if ( ! hasStarted ) {
+						hasStarted = true;
+						node.style.pointerEvents = 'none';
+					}
+					node.style.top = `${ e.clientY * inverted - originY }px`;
+					node.style.left = `${ e.clientX * inverted - originX }px`;
+				}
 
 				function end() {
 					ownerDocument.removeEventListener( 'dragover', over );
 					ownerDocument.removeEventListener( 'dragend', end );
-					domNode.remove();
+					node.style.transform = '';
+					node.style.transformOrigin = '';
+					node.style.transition = '';
+					node.style.zIndex = '';
+					node.style.position = '';
+					node.style.top = '';
+					node.style.left = '';
+					node.style.width = '';
+					node.style.pointerEvents = '';
+					node.style.margin = '';
+					node.style.opacity = '';
+					node.style.backgroundColor = '';
```

Comment:
At this point, we may need to restore any styles that were originally applied to the block. For example, if you drop a block with a margin or background color, the style will disappear.

https://github.com/user-attachments/assets/1a8d7318-a7aa-457a-81a4-f5f18e5e0f44